regionServiceClient/lib/cloudregister/registerutils.py

```diff
@@ -311,11 +318,15 @@ def get_current_smt():
         return
     # Verify that this system is also in /etc/hosts and we are in
     # a consistent state
-    smt_ip = smt.get_ip()
+    smt_ipv4 = smt.get_ipv4()
+    smt_ipv6 = smt.get_ipv6()
```

Comment:
Should this and the re.search be conditional? From below it looks like get_ipv6() could yield None which would mean searching for 'None\s'.

Reply:
Yes, smt_ipv6 can be None. The trade of is to spend CPU cycles searching for "None" which should never succeed, or spending those cycles in an additional branching instruction with additional book keeping requirements and more code. The current logic is:

if ipv4 or ipv6 and fqdn are in the hosts file all is well

this would have to change into

if ipv4 not found and ipv6 has a value look for ipv6, if ip address entry is found look for fqdn entry

I don't see the benefit of the extra conditional w.r.t. clarity of logic nor expressiveness of the written code.